block_sharing_cart.php

```diff
@@ -205,8 +206,12 @@ public function get_content() {
         return $this->content = (object) array('text' => $html, 'footer' => $footer);
     }
 
-    private function is_activity_not_in_section(int $section_id, cm_info $activity): bool {
-        return $section_id !== $activity->get_section_info()->section;
+    private function is_activity_in_section(int $section_id, int $section, \cm_info $activity): bool {
+        if ($section === 0 && $section_id == 0) {
+            return $activity->get_section_info()->section === 0;
+        }
+
+        return ($section === $activity->get_section_info()->section && $activity->get_section_info()->section !== 0) || $section_id == $activity->get_section_info()->id;
```

Comment:
`$activity->get_section_info()` call should be moved to a variable.
The repeated calls makes the line was to long, and might not be good for performance.